Sensors.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Sensors.h"
+#include <Adafruit_MAX31855.h>
+#include <SPI.h>
+#include <math.h>
+
+Sensors::Sensors(int sensorIndex){
+	pP_Abs = (P_AbsPinout + sensorIndex);
+	pP_Gauge = (P_GaugePinout + sensorIndex);
+	pTherm = (ThermPinout + sensorIndex);
+	pFlow = (FlowPinout + sensorIndex);
+	pTFlow = &TFlowPinout;
+	pO2 = &O2Pinout;
+	pO2Therm = &O2ThermPinout;
+	pHeaterCurrent = (HeaterCurrentPinout + sensorIndex);
+	Adafruit_MAX31855 thermocouple(*pTherm);

```

Comment:
Also note. You define "thermocouple" within the constructor. As soon as the constructor exits/finishes, the thermocouple object gets destroyed.